src/main/java/net/neoforged/neoforge/common/util/AttributeUtil.java

```diff
@@ -0,0 +1,356 @@
+            double amt = base;
+
+            // Compute the base value including merged modifiers if merging is enabled
+            if (NeoForgeMod.shouldMergeAttributeTooltips()) {
+                for (AttributeModifier modif : baseModif.children) {
+                    switch (modif.operation()) {
+                        case ADD_VALUE:
+                            base = amt = amt + modif.amount();
+                            break;
+                        case ADD_MULTIPLIED_BASE:
+                            amt += modif.amount() * base;
+                            break;
+                        case ADD_MULTIPLIED_TOTAL:
+                            amt *= 1 + modif.amount();
+                            break;
+                    }
+                }
+            }
+
+            double baseBonus = attr.value().getBonusBaseValue(stack);
+            amt += baseBonus;
+
+            boolean isMerged = NeoForgeMod.shouldMergeAttributeTooltips() && (!baseModif.children.isEmpty() || baseBonus != 0);
+            MutableComponent text = attr.value().toBaseComponent(amt, entityBase, isMerged, ctx.flag());
+            tooltip.accept(Component.literal(" ").append(text).withStyle(isMerged ? ChatFormatting.GOLD : ChatFormatting.DARK_GREEN));
+            if (ctx.flag().hasShiftDown() && isMerged) {
+                // Display the raw base value, and then all children modifiers.
+                text = attr.value().toBaseComponent(rawBase, entityBase, false, ctx.flag());
+                tooltip.accept(AttributeUtil.list().append(text.withStyle(ChatFormatting.DARK_GREEN)));
+                for (AttributeModifier modifier : baseModif.children) {
+                    tooltip.accept(AttributeUtil.list().append(attr.value().toComponent(modifier, ctx.flag())));
+                }
+                if (baseBonus > 0) {
+                    attr.value().addBonusTooltips(stack, tooltip, ctx.flag());
+                }
+            }
+        }
+
+        for (Holder<Attribute> attr : modifierMap.keySet()) {
+            // Skip attributes who have already been processed during the base modifier stage
+            if (NeoForgeMod.shouldMergeAttributeTooltips() && baseModifs.containsKey(attr)) {
+                continue;
+            }
+
+            Collection<AttributeModifier> modifs = modifierMap.get(attr);
+            // Initiate merged-tooltip logic if we have more than one modifier for a given attribute.
+            if (NeoForgeMod.shouldMergeAttributeTooltips() && modifs.size() > 1) {
+                double[] sums = new double[3];
+                boolean[] merged = new boolean[3];
+                Map<Operation, List<AttributeModifier>> shiftExpands = new HashMap<>();
+
+                for (AttributeModifier modifier : modifs) {
+                    if (modifier.amount() == 0) continue;
+                    if (sums[modifier.operation().ordinal()] != 0) merged[modifier.operation().ordinal()] = true;
```

Comment:
Can we just retrieve the ordinal once and store it in a variable so that this code is a bit more readable?